src/main/java/io/vertx/core/json/Json.java

```diff
@@ -72,12 +74,26 @@ public static String encodePrettily(Object obj) throws EncodeException {
   }
 
   public static <T> T decodeValue(String str, Class<T> clazz) throws DecodeException {
+    return decodeValue(str, clazz, true);
+  }
+
+  public static <T> T decodeValue(String str, Class<T> clazz, boolean failOnUnknownProperties) throws DecodeException {
+    ObjectReader reader = mapper.readerFor(clazz);
+
+    if (failOnUnknownProperties) {
+      reader = reader.with(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    } else {
+      reader = reader.without(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    }
+
```

Comment:
Just a thought, wouldn't it create a new reader everytime ? 
Shouldn't we try to use the object mapper if `failOnUnknownProperties` is true (there is a rish that the mapper has been configured otherwise).